### ISSUE TITLE
Long lived operations & UiComponents

### ISSUE THREAD

**Issue by manueldidonna** [julianomoraes/componentizationArch#5]
I have a doubt about components lifecycle placed in a fragment(or activity).

UiComponents are destroyed and created accordingly to the fragment's lifecycle. So on each configuration changes we created a new component instance. 

Where should I put long lived operations (network request, ...)? Should I use a viewmodel? 

I'm also thinking on a mechanism to save the state of those components that have it.

**Comment by julianomoraes** [comment]
Yes, you can have a ViewModel owning the UI Components. But keep in mind they will not be re-created on configuration changes, so they will have to handle rotations gracefully. 

Keep in mind the subscription of the components is what keeps them alive (not Gced), if you don't hold a ref for them in the Frag. The subscription can also be hooked up with the ViewModel Lifecycle.

**Comment by manueldidonna** [comment]
I decided to opt for a different approach. The ViewModel doesn't own the components (maybe it could intercept their events but I consider it unnecessary)

I create two kinds of components:

• EphemeralComponents that don't survive to configuration changes or process death

• StatefulComponents that can survive both configuration changes (keepComponentInstance flag) and process death (saving their states). I have my own custom manager.

I also need to use a viewpager, that's why I created a LazyInflatedUIView, but some behaviors have yet to be defined.

When everything is ready I will release the code, hoping to get some reviews and feedback. I'm also creating an app to explore the advantages of this architecture and I'm trying to find out any pitfalls.
